dexios/src/subcommands/header_key.rs

```diff
@@ -140,7 +142,112 @@ pub fn change_key(input: &str, key_old: &Key, key_new: &Key) -> Result<()> {
+
+            // TODO(brxken128): convert this to the `decrypt_master_key` contained in #140
+            for (i, keyslot) in keyslots.iter().enumerate() {
+                let hash_start_time = Instant::now();
+                let key_old = keyslot
+                    .hash_algorithm
+                    .hash(raw_key_old.clone(), &keyslot.salt)?;
+                let hash_duration = hash_start_time.elapsed();
+                success!(
+                    "Successfully hashed your old key [took {:.2}s]",
+                    hash_duration.as_secs_f32()
+                );
+
+                let cipher = Ciphers::initialize(key_old, &header.header_type.algorithm)?;
+
+                let master_key_result =
+                    cipher.decrypt(&keyslot.nonce, keyslot.encrypted_key.as_slice());
+
+                if master_key_result.is_err() {
+                    continue;
+                }
+
+                let mut master_key_decrypted = master_key_result.unwrap();
+                let len = MASTER_KEY_LEN.min(master_key_decrypted.len());
+                master_key[..len].copy_from_slice(&master_key_decrypted[..len]);
+                master_key_decrypted.zeroize();
+
+                index = i;
+
+                drop(cipher);
+                break;
+            }
+
+            if master_key == [0u8; MASTER_KEY_LEN] {
+                return Err(anyhow::anyhow!("Unable to find a match with the key you provided (maybe you supplied the wrong key?)"));
+            }
+
+            let master_key = Protected::new(master_key);
+
+            match key_new {
+                Key::Generate => info!("Generating a new key"),
+                Key::User => info!("Please enter your new key below"),
+                Key::Keyfile(_) => info!("Reading your new keyfile"),
+                Key::Env => (),
+            }
+            let raw_key_new = key_new.get_secret(&PasswordState::Validate)?;
+            let salt_new = gen_salt();
+            let hash_start_time = Instant::now();
+            // TODO(brxken128): allow using argon2id/balloon/inherit
+            let key_new = keyslots[index]
+                .hash_algorithm
+                .hash(raw_key_new, &salt_new)?;
+            let hash_duration = hash_start_time.elapsed();
+            success!(
+                "Successfully hashed your new key [took {:.2}s]",
+                hash_duration.as_secs_f32()
+            );
+
+            let cipher = Ciphers::initialize(key_new, &header.header_type.algorithm)?;
+
+            let master_key_nonce_new = gen_nonce(&header.header_type.algorithm, &Mode::MemoryMode);
+            let master_key_result =
+                cipher.encrypt(&master_key_nonce_new, master_key.expose().as_slice());
+
+            drop(master_key);
+
+            let master_key_encrypted = master_key_result
+                .map_err(|_| anyhow::anyhow!("Unable to encrypt your master key"))?;
+
+            let mut master_key_encrypted_array = [0u8; 48];
```

Comment:
We already have `dexios_core::primitives::ENCRYPTED_MASTER_KEY_LEN;`

Reply:
A local commit updated to that, maybe I forgot to push :)